### PR TITLE
Restaurant page

### PR DESCRIPTION
**Pull request by erezfree29**
Restaurant page
Done in accordance with the Udin project Restaurant exercise.
skills applied 
1)NPM
2)Webpack
3)Javascript
4)Functions exporting.
5)Css
Only one basic Html file is used while the content is written and wiped out using purely JavaScript

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+-   NPM
+-   WEBPACK
+-   HTML
+-   CSS
+-   VScode
+-   Lots of love :heart:
+
+### Installation
+
+In order to run the the project on your local machine, please run the following commands on your terminal:
```
Comment (rindrajosia) [inline README.md]: In order to run `the project` on your local machine, please run the following commands on your terminal:

File: src/scripts/contact.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+function contact() {
```
Comment (rindrajosia) [inline src/scripts/contact.js]: usage of arrow functions instead of the function keyword

File: src/scripts/description.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+function description() {
```
Comment (rindrajosia) [inline src/scripts/description.js]: usage of arrow functions instead of the function keyword

File: src/scripts/header.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+function headline() {
```
Comment (rindrajosia) [inline src/scripts/header.js]: usage of arrow functions instead of the function keyword

File: src/scripts/home.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+function home() {
```
Comment (rindrajosia) [inline src/scripts/home.js]: usage of arrow functions instead of the function keyword

File: src/scripts/index.js [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+import header from './header';
+import pictureDiv from './picture';
+import description from './description';
+
+const menuButton = document.querySelector('.menu');
+const homeButton = document.querySelector('.home');
+const contactButton = document.querySelector('.contact');
+const content = document.querySelector('.content');
+
+homeButton.onclick = function () {
```
Comment (rindrajosia) [inline src/scripts/index.js]: usage of arrow functions instead of the function keyword

File: src/scripts/index.js [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    contactContent.parentNode.removeChild(contactContent);
+    document.body.appendChild(content);
+  }
+  if (menuContent !== null) {
+    menuContent.parentNode.removeChild(menuContent);
+    document.body.appendChild(content);
+  }
+};
+
+contactButton.onclick = function () {
```
Comment (rindrajosia) [inline src/scripts/index.js]: usage of arrow functions instead of the function keyword

File: src/scripts/index.js [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+</div>
+  `;
+  container.appendChild(usheader);
+  const searchContent = document.querySelector('.contact_content');
+  if (searchContent === null) {
+    document.body.appendChild(contactcontent);
+  }
+};
+
+menuButton.onclick = function () {
```
Comment (rindrajosia) [inline src/scripts/index.js]: usage of arrow functions instead of the function keyword

File: src/scripts/menu.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+function menu() {
```
Comment (rindrajosia) [inline src/scripts/menu.js]: usage of arrow functions instead of the function keyword

File: src/scripts/navbar.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+function navBar() {
```
Comment (rindrajosia) [inline src/scripts/navbar.js]: usage of arrow functions instead of the function keyword

File: src/scripts/picture.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import Restaurant from '../images/Restaurant.jpg';
+
+function pictureDiv() {
```
Comment (rindrajosia) [inline src/scripts/picture.js]: usage of arrow functions instead of the function keyword

File: src/scripts/index.js [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+    <div class="row">Name: Snopy the dog</div>
+    <div class="row">Positon: Chef</div>
+    <div class="row">Email: wofwof@unavilable.com</div>
+  </div>
+  <div class="contactp third">
+  <div class="row">Name: Mr Bin</div>
+  <div class="row">Positon: Waiter</div>
+  <div class="row">Email: stuckinthe90's@unavilable.com</div>
+</div>
+  `;
```
Comment (rindrajosia) [inline src/scripts/index.js]: The contents of each tab are stored in a module.

File: src/scripts/index.js [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+          <div>Price: $9</div>
+        </div>
+        <div class="dish">
+          <div>Dish: Sushi</div>
+          <div>cousine: Japanes</div>
+        <div>Price: $13</div>
+      </div> 
+    </div>
+  </div>
+  `;
```
Comment (rindrajosia) [inline src/scripts/index.js]: The contents of each tab are stored in a module.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- It uses Webpack :+1: 

To improve :factory_worker: [REQUIRED]:

- [ ] Files not related to the project should not be committed and added to .gitignore (e.g. .vscode and draft.html) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [ ] It includes a rule for the node_modules directory on .gitignore ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/restaurant_page/gitignore_requirement_pathwright.png) from [Odin](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622983/))
- [ ] As you are using ES6 syntax, please use `arrow functions `instead of `the function keyword`  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/restaurant_page/es6_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622983/))
- [ ] The contents of each tab are stored in a module. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/restaurant_page/tab_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/restaurant-page))
- [ ] Please check your readme by using Grammarly because it is the first thing that the user reads

Nice to have [OPTIONAL] :

- [ ] 

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @erezfree29  👋

**You've done pretty well here fixing a few issues raised in the previous review. 👌**

However, let's make a couple of changes.

To highlight: 
- Fix the module export in the app.
- Use es6 arrow functions.

- Please refer to other comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/scripts/homebuttonpage.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+      contactContent.parentNode.removeChild(contactContent);
+      document.body.appendChild(content);
+    }
+    if (menuContent !== null) {
+      menuContent.parentNode.removeChild(menuContent);
+      document.body.appendChild(content);
+    }
+  });
+}
+export default homebuttonpage();
```
Comment (mikenath223) [inline src/scripts/homebuttonpage.js]: - [ ] This isn't the right way to export a module. Do this instead: `export default homebuttonpage;`. **Fix all other exported modules also.**

File: src/scripts/index.js [hunk 15]
```diff
@@ -9,154 +10,6 @@ import contact from './contact';
...
-  const searchMenuContent = document.querySelector('.menu_content');
-  if (searchMenuContent === null) {
-    document.body.appendChild(menuContent);
-  }
-};
-
-//
+import homeButtonPage from './homebuttonpage';
+import contactButtonPage from './contactbuttonpage';
+import menuButtonPage from './menubuttonpage';
```
Comment (mikenath223) [inline src/scripts/index.js]: - [ ] After importing the modules you can call them this way: `homeButtonPage()`, As they are functions. Then simply write logic here to call a function when the particular nav item is clicked.

File: src/scripts/menubuttonpage.js [hunk 16]
```diff
@@ -0,0 +1,93 @@
+function menuButtonPage() {
+  const menuButton = document.querySelector('.menu');
+  menuButton.addEventListener('click', () => {
+    const content = document.querySelector('.content');
+    const contactContent = document.querySelector('.contact_content');
+    if (content !== null) {
+      content.parentNode.removeChild(content);
```
Comment (mikenath223) [inline src/scripts/menubuttonpage.js]: - [ ] Please change this and **all** other regular functions to es6 arrow function syntax.

File: src/scripts/homebuttonpage.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
+function homebuttonpage() {
+  const content = document.querySelector('.content');
+  const homeButton = document.querySelector('.home');
+  homeButton.addEventListener('click', () => {
+    const contactContent = document.querySelector('.contact_content');
```
Comment (mikenath223) [inline src/scripts/homebuttonpage.js]: - [ ] This isn't the right way to export a module. Do this instead: `export default homebuttonpage;`. **Fix all other exported modules also.**

**Comment by mikenath223** [comment]
- There was a large subset of changes so kindly refer to the comments above as the review. Thanks

**Review by mikenath223 — CHANGES_REQUESTED** [review]
'

**Review by mke2111 — APPROVED** [review]
# 😄 🥇  STATUS: APPROVED  🥇 👏🏽 

## Hey Happy Coder, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval. 👍🏽 

### **_Well Done_** 💯 

    1. The ES6 syntax is applied appropriately



### You may now merge to the develop branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

<hr>

### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Pull request merged**